test/index.spec.ts

```diff
@@ -137,452 +137,540 @@ test('encryptor:encrypt & decrypt with wrong password', async ({ page }) => {
-    { password, data: JSON.stringify(decryptable) },
-  );
+      expect(decryptedData).toStrictEqual(expectedData);
+    });
+
+    test('encryptor:decrypt encrypted data using key derived from wrong password', async ({
+      page,
+    }) => {
+      const wrongPassword = 'a wrong password';
+
+      await expect(
+        page.evaluate(
+          async (args) => {
+            const key = await window.encryptor.keyFromPassword(
+              args.wrongPassword,
+              args.salt as string,
+              false,
+              args.encryptedPayload.keyMetadata,
+            );
+            return await window.encryptor.decryptWithKey(
+              key,
+              args.encryptedPayload,
+            );
+          },
+          {
+            encryptedPayload: testEncryptedData,
+            salt: testEncryptedData.salt,
+            wrongPassword,
+          },
+        ),
+      ).rejects.toThrow('Incorrect password');
+    });
+
+    test('encryptor:importKey generates valid CryptoKey', async ({ page }) => {
+      const isKey = await page.evaluate(
+        async (args) => {
+          const encryptionKey = await window.encryptor.importKey(
+            args.SAMPLE_EXPORTED_KEY,
+          );
+          return encryptionKey instanceof CryptoKey;
+        },
+        { SAMPLE_EXPORTED_KEY },
+      );
+      expect(isKey).toBe(true);
+    });
+
+    test('encryptor:exportKey generates valid CryptoKey string', async ({
+      page,
+    }) => {
+      const keyString = await page.evaluate(
+        async (args) => {
+          const key = await window.encryptor.importKey(
+            args.SAMPLE_EXPORTED_KEY,
+          );
+          return await window.encryptor.exportKey(key);
+        },
+        { SAMPLE_EXPORTED_KEY },
+      );
+      expect(keyString).toStrictEqual(SAMPLE_EXPORTED_KEY);
+    });
+
+    test('encryptor:encryptWithDetail and decryptWithDetail provide same data', async ({
+      page,
+    }) => {
+      const password = 'a sample passw0rd';
+      const data = { foo: 'data to encrypt' };
+
+      const { vault } = await page.evaluate(
+        async (args) =>
+          await window.encryptor.encryptWithDetail(args.password, args.data),
+        { data, password },
+      );
 
-  expect(JSON.stringify(decryptedResult.vault)).toStrictEqual(
-    JSON.stringify(newData),
-  );
-});
+      const decryptedDetail = await page.evaluate(
+        async (args) =>
+          await window.encryptor.decryptWithDetail(args.password, args.data),
+        { data: vault, password },
+      );
 
-test('encryptor:encryptWithKey works with decryptWithKey', async ({ page }) => {
-  const password = 'a sample passw0rd';
-  const startingData = { foo: 'data to encrypt' };
+      expect(JSON.stringify(decryptedDetail.vault)).toStrictEqual(
+        JSON.stringify(data),
+      );
+    });
+
+    test('encryptor:decryptWithKey provide same data when using exported key from encryptWithDetail', async ({
```

Comment:
This test doesn't rely on either `sampleEncryptedData` or `oldSampleEncryptedData`. Should it be inside the loop?